javacs-lab12/src/com/flatironschool/javacs/JedisIndex.java

```diff
@@ -97,9 +99,31 @@ public void add(String term, TermCounter tc) {
 		return map;
 	}
 
+
+	/**
+	 * Looks up a term and returns a map from URL to TF-IDF count.
+	 *
+	 * @param term
+	 * @return Map from URL to count.
+	 */
+
+	 //Can make this faster through using the "GET COUNTS FASTER" method
+	public Map<String, Integer> getCountsTFIDF(String term) {
+		StopWords stop = new StopWords();
+		Map<String, Integer> map = new HashMap<String, Integer>();
+		Set<String> urls = getURLs(term);
+		for (String url: urls) {
+			if(stop.getTFIDF(this, url, term) == null) { return null; }
```

Comment:
right here the same method getTFIDF is called twice, which as stated returns the product of tf and idf. but in calling it twice it calls the inversedocfreq method twice, which iterates through the url set to check if there is an instance of the search term in each url. might it be better to just call the method once and store it as a variable? it would not really affect the execution much in our case but idk which is better "theoretically"/for potentially larger samples